src/group_theory/free_product.lean

```diff
@@ -151,6 +151,36 @@ lemma of_left_inverse [decidable_eq ι] (i : ι) :
 lemma of_injective (i : ι) : function.injective ⇑(of : M i →* _) :=
 by { classical, exact (of_left_inverse i).injective }
 
+lemma lift_range_subset_submonoid {N} [monoid N] (f : Π i, M i →* N) {s : submonoid N}
+  (h : ∀ i, set.range (f i) ⊆ s) : set.range (lift f) ⊆ s :=
```

Comment:
How about:
```suggestion
  (h : ∀ i, (f i).mrange ≤ s) : (lift f).mrange ≤ s :=
```

Reply:
I tend to agree, but again, this is to match https://leanprover-community.github.io/mathlib_docs/group_theory/free_group.html#free_group.lift.range_subset - should I change that as well?

(It's really mostly just a auxillary lemma for the next step, so maybe fine as it is.)